### DISCUSSION TITLE
How do I make my ragdoll not turn into a puddle?

### DISCUSSION THREAD

**Discussion by Gamepro5** [godot-jolt/godot-jolt#825] [Questions]
https://github.com/godot-jolt/godot-jolt/assets/23440295/a6d2c8ad-e188-4422-9eed-23e60a7468dd

I adjusted the size of the physical bones, I am not sure on what joins need to be what value though because the joints don't have much documentation on Godot's end and I don't how to corelate real human joins to Godot joints.

**Comment by mihe** [comment]
Honestly, I haven't dealt much with Godot's ragdolls, or any ragdolls for that matter. They're not implemented from the physics server, but rather lives purely on the scene side as a sort of higher-level node, so I've mostly just made sure that the few example scenes I've seen behave somewhat fine.

I would probably reach for `ConeTwistJoint3D` myself, since it maps pretty well to how our actual joints behave, in that they can only twist a certain amount and swing a certain amount.

In any case, since this isn't Jolt-specific per se, I'd probably recommend asking around in the more general Godot communities.

**Comment by V1LLz** [comment]
I recommend that you take all the PhysicalBones3D and give them a 6DOFJoint articulation, except for the parent bone, which is usually the hip. For this last one, set it to None articulation.  Then, adjust the angular limits for each axis of each physical bone. One thing I can tell you from my experience is that in the gizmo, the angular limits are inverted. So, if you want the character to have a range, for example, of 90° on the x-axis, and the gizmo shows that this range is forward, then it will go backward (it sounds a bit strange, but it's a bug that has been in Godot for a while 😅). Here's a link to the issue that can help you understand it a bit better: [godotengine/godot/issues/84468]

Just like in the documentation, remove all unnecessary bones such as the hand bones, neck bone, a MASTER bone, and others.
